### PR TITLE
Debian Package

### LINKED ISSUE #6: deb package
any volunteer?

### PR DESCRIPTION
**Pull request by FloydOtsoko**
- Packaging for Debian with workflow to create release with package on push of tags starting with `v*`
- Update of README.md file
- ~~Regolith3 support according to~~ [~~Regolith 3.0 Release Notes~~](https://regolith-desktop.com/docs/reference/Releases/regolith-3.0-release-notes/)

Closes #6

### CODE REVIEW

**Comment by FloydOtsoko** [comment]
So as far as my newly found knowledge about debian packaging with github workflows goes, I believe that simply merging this pull request will most likely break the workflow. 

The Reson is this part of the action code:
```yaml
      - name: Release the Package
        uses: softprops/action-gh-release@v1
        with:
          name: i3-workspace-names-daemon
          files: /home/runner/work/i3-workspace-names-daemon/*.deb
          token: ${{ secrets.release }}
```
to release the package via the action in order to make the deb file available, I had to setup a secret with github so that the repository allows the changes.

I am neither able to create such token for the upstream repo (most likely because of permissions) nor can I do this for the newly created /deb repo (thus I am still using my fork)

Open for discussion on how to move forward

**Comment by CastixGitHub** [comment]
Hi
- Can you please also document how to build the deb yourself without depending/trusting on github? (given the benefit of having a deb package is uniformity with system management here)
- The license chosen by cboddy was MIT (Please avoid picking licenses with terms that are more restrictive than the packaged work)
- The copyright file states that `debian/*` files are from cboddy, meanwhile the root ones are unknown
- Is there a reason to place a github-noreply email as mantainer? looks bad to me
- The package suggests `fonts-font-awesome` then i wonder how should the version matching work (between the fa_icons.py file and the installed font (see #10 )) (that's another reason to have such a deb package)

**Comment by FloydOtsoko** [comment]
Hello,
sorry that my response took so long.

Tl;dr: 
- Closed PR
- setup email
- will rework deb package and make proper doc/license/©
- will split PR (extra PR for regolith3 and FA 6.5.1)
- `fonts-font-awesome` works with older icons (FA <= 4.7.0)


Long Story:
So, I did some cleanup today and temporarily closed the PR. I made a new one for regolith 3 and FA update #15  and will continue the debian package on a different branch for now.

Comment answers:
1) I will go through everything again and create a documentation along the way. Since I have suffered a dying hard drive recently and not backed up my (very little) documentation I had (yes, lesson learned) this will take some time.
2) Sorry for this error. This happened most likely due to my limited knowledge of debian packaging. I will make sure that the license and copyright are setup correctly.
3)  same as 2.
4) an attempt of keeping some privacy tbh. However, I have setup a new email and will use it for the commits to come. 
5) moved to #10 

So, moving forward I created new issues/PRs for regolith3 support as well as an update for `fa_icons.py` to fontawesome `6.5.1`
When its done I will reopen this PR and continue developing the debian package, but I think this will take some time to complete. I have an increased workload IRL as well as a (faulty and incomplete but) working debian package holding me back.

**Comment by CastixGitHub** [comment]
Hi @FloydOtsoko i surrended and resurrected my github account. (I also came back to i3 (had gnome for a while at work. and then months of bare ttys, launched [twb](https://framagit.org/castix/twb)))
So i merged a couple of PR from you, the deb repo is still empty, should i delete it?

**Comment by FloydOtsoko** [comment]
as mentioned [in this comment](https://github.com/i3-workspace-names-daemon/i3-workspace-names-daemon/pull/12#issuecomment-1779439974) the deb repo is not really useful for me in its current state.

Either I get enough permissions to create gitlab secrets, or you can delete it and I will use my fork to propose PRs

**Comment by CastixGitHub** [comment]
Consider this fully merged (you did force push so this PR has 0 commits and committers report graph doesn't reflect,)

(And i managed not to have to setup any secret at all)
